src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -989,20 +989,33 @@ bool clang_cpp_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
     const clang::LambdaExpr &lambda_expr =
       static_cast<const clang::LambdaExpr &>(stmt);
 
-    if (lambda_expr.capture_size() > 0)
-    {
-      log_error("ESBMC does not support lambdas with captures for now");
-      return true;
-    }
-
     get_struct_union_class(*lambda_expr.getLambdaClass());
 
     // Construct a new object of the lambda class
     typet lambda_class_type;
     if (get_type(
           *lambda_expr.getLambdaClass()->getTypeForDecl(), lambda_class_type))
       return true;
-    new_expr = gen_zero(pointer_typet(lambda_class_type));
+
+    if (lambda_expr.capture_size())
+    {
+      const struct_union_typet &this_type =
+        to_struct_union_type(ns.follow(lambda_class_type));
+      exprt sym("struct", this_type);
+      for (auto it = lambda_expr.capture_init_begin();
```

Comment:
Can you not use `capture_init` here like so?
```cpp
for (const auto &init : lambda_expr.capture_inits())
      {
        exprt init_expr;
        if (get_expr(*init, init_expr))
          return true;

        sym.move_to_operands(init_expr);
      }
```
 https://clang.llvm.org/doxygen/classclang_1_1LambdaExpr.html#acbad4b6db44728efe65914d34d36a91c here instead?